src/extension.ts

```diff
@@ -62,10 +62,17 @@ export function activate(context: ExtensionContext) {
 
     // Register support for unicode input.
     (async () => {
-        const translations: any = await loadJsonFile.default(context.asAbsolutePath('translations.json'));
+        const translations : any = await loadJsonFile.default(context.asAbsolutePath('translations.json'));
+        const modes = [LEAN_MODE];
+        const input_markdown = workspace.getConfiguration('lean.input').get('markdown');
+        if (input_markdown) {modes.push(MARKDOWN_MODE);}
+        const hover_providers = modes.map(mode =>   
+            languages.registerHoverProvider(mode, new LeanInputExplanationHover(translations))
+        );
+
         context.subscriptions.push(
-            languages.registerHoverProvider(LEAN_MODE, new LeanInputExplanationHover(translations)),
-            new LeanInputAbbreviator(translations, LEAN_MODE));
+            ...hover_providers,
```

Comment:
I hope that we can get by with a single `LeanInputExplanationHover` if we register it for `{}: DocumentFilter`.

Reply:
I couldn't get this to work :(

Reply:
I'm now doing `languages.registerHoverProvider(inputLanguages, new LeanInputExplanationHover(translations));` in `extension.ts` where `inputLanguages` is retieved from settings and is eg `["lean", "markdown"]`. So you have to reload if you change `lean.input.languages` and want to get hover hints for the new language.

Reply:
Only the hovers though

Reply:
Ok, it was worth a try.